datacube/model/__init__.py

```diff
@@ -574,10 +582,11 @@ def __eq__(self, other):
         if not isinstance(other, GridSpec):
             return False
 
-        return (self.crs == other.crs
-                and self.tile_size == other.tile_size
-                and self.resolution == other.resolution
-                and self.origin == other.origin)
+        crs_result = self.crs == other.crs
```

Comment:
Can we revert this change please? Original check short circuits computation, while this one forces each one to be computed and forces creation and choosing names for temporary variables. We should probably re-order so that expensive comparison `crs==crs` is last and cheaper ones are first.

Reply:
I reverted the changes.